hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNodeRpcServer.java

```diff
@@ -1899,7 +1907,29 @@ private void verifySoftwareVersion(DatanodeRegistration dnReg)
     }
   }
 
-  private static String getClientMachine() {
+  private String getClientMachine() {
+    if (ipProxyUsers != null) {
+      // Get the real user (or effective if it isn't a proxy user)
```

Comment:
To avoid having a long getClientMachine(), it may be better to have a separate private to get the real user.

Reply:
It would also help to have a comment with the example of what the key you are passing looks like.
Potentially could have its own unit test.